mantidproject/mantid
# Make data processor interface presenters independent of Qt

### Description of work
As explained in #37290 , this PR refactors the data processor classes that are inherited form the inelastic tab, to make them independent of qt. In order to do that, the algorithms from the derived presenter classes are called by using an algorithm runner implemented with the subscriber pattern. Now the data processor classes hold an algorithm runner, and a notifier class. 
Although the presenters are created using unique_ptrs to `IDataProcessor` interfaces. The data processor class still needs to be derived form `InelasticTab`. This PR is a first step on getting rid of this dependency, as we want to eventually remove the `InelasticTab` class from the data processor interface. 
#### Summary of work




Fixes #37290. 




#### Further detail of work


### To test:
- Make sure there are no warnings related to this pr on the build. 
- Follow data processor manual test: https://developer.mantidproject.org/Testing/Inelastic/DataProcessorTests.html



*No release notes, internal change*



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
